### ISSUE TITLE
Projects with the sequence "\b" in any string input give a 500 error when saving

### ISSUE THREAD

**Issue by adroitwhiz** [LLK/scratch-www#3497]
If any string input/argument in any block in a project contains the substring "\b" anywhere, attempts to save the project to the Scratch server will result in an HTTP 500 error.

Other backslash sequences ("\n", "\t", etc.) seem to work fine--I've tested all letters of the alphabet (in lowercase) and only "\b" causes a 500 error.

Downloading the project as a .sb3 works fine, so this appears to be a website issue which is why I'm filing this bug against scratch-www (there's no public repo for the Scratch backend).

### Steps to Reproduce

1. Type a string containing "\b" into any text input:
![image](https://user-images.githubusercontent.com/25993062/68028211-aecf0500-fc8a-11e9-8e5d-e8fd348820c1.png)

2. Observe the 500 error when saving:
![image](https://user-images.githubusercontent.com/25993062/68028238-c27a6b80-fc8a-11e9-9eb8-a3b32d00d0a1.png)
![image](https://user-images.githubusercontent.com/25993062/68028251-ce662d80-fc8a-11e9-9f04-67ea6396072c.png)

### Operating System and Browser
Firefox 69.0.1 on Fedora 30 (though this should occur on every browser)

**Comment by apple502j** [comment]
A bit of history here.

Disapperaing Script Bug was a major bug in early Scratch 3.0. Some weird macOS Japanese IME added ASCII Backspace to some places where they shouldn't be. In Scratch 3.0, scripts are made into XML (for use in scratch-blocks). Of course raw Backspace can't be inside XML. That made XML parsing error, causing scripts to "disappear".

A hacky fix to scratch-parser, which removed Backspace before parsing raw string as project JSON, fixed the problem. But, the string `\b` in Scratch, which is `\\b`, is now made into `\`! This is how it's caused.

TL;DR:
Duplicate of LLK/scratch-parser#56

**Comment by Scratch-Cat** [comment]
Try using: say ( join [\] [b] )

**Comment by BryceLTaylor** [comment]
I'm not sure this is a duplicate of LLK/scratch-parser#56.  That one is describing loading behavior and this issue is describing saving behavior.  They are related, but I think it's worth keeping both issues open.

Interestingly, I'm unable to reproduce the inability to save if the \b is in the middle of a string (on mac chrome).

**Comment by apple502j** [comment]
@BryceLTaylor If I remember correctly Scratch actually validates Scratch project in some way in the backend (not sure if it's scratch-parser) - if it's the case it's duplicate because the main cause is the same. Also, "saving" does not include saving locally - only for saving to Scratch server.

Can repro: Firefox 78 on Windows 10
![image](https://user-images.githubusercontent.com/33279053/87174411-19543c80-c312-11ea-9bea-981475053589.png)

**Comment by benjiwheeler** [comment]
Was able to repro: Chrome 89 on Mac

![image](https://user-images.githubusercontent.com/3431616/103930777-45981c80-50ed-11eb-9127-702a53425415.png)

**Comment by benjiwheeler** [comment]
I brought this to the attention of the group working on revisiting save alerts; I imagine it will require work on scratch-parser and/or scratch-gui; scratch-parser has this issue filed: https://github.com/LLK/scratch-parser/issues/56

**Comment by Scratch-Cat** [comment]
Hello, could you tell me why I'm now unable to access Scratch?




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Benjamin Wheeler"<notifications@github.com&gt;; 
发送时间: 2021年1月8日(星期五) 凌晨2:36
收件人: "LLK/scratch-www"<scratch-www@noreply.github.com&gt;; 
抄送: "136"<2398574290@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [LLK/scratch-www] Projects with the sequence &quot;\b&quot; in any string input give a 500 error when saving (#3497)





 
Was able to repro: Chrome 89 on Mac

**Comment by benjiwheeler** [comment]
@Scratch-Cat email me outside this issue please: ben@scratch.mit.edu

**Comment by benjiwheeler** [comment]
Just clarifying that this has a proposed solution PR, and we just need to review it
